connectors/src/lib/temporal.ts

```diff
@@ -76,32 +78,52 @@ export async function getTemporalWorkerConnection(): Promise<{
 export async function getConnectorId(
   workflowRunId: string
 ): Promise<ModelId | null> {
-  if (!WORKFLOW_ID2CONNECTOR_ID_CACHE[workflowRunId]) {
+  if (!CONNECTOR_ID_CACHE[workflowRunId]) {
     const client = await getTemporalClient();
-    const workflowHandle = await client.workflow.getHandle(workflowRunId);
+    const workflowHandle = client.workflow.getHandle(workflowRunId);
     const described = await workflowHandle.describe();
     if (described.memo && described.memo.connectorId) {
       if (typeof described.memo.connectorId === "number") {
-        WORKFLOW_ID2CONNECTOR_ID_CACHE[workflowRunId] =
-          described.memo.connectorId;
+        CONNECTOR_ID_CACHE[workflowRunId] = described.memo.connectorId;
       } else if (typeof described.memo.connectorId === "string") {
-        WORKFLOW_ID2CONNECTOR_ID_CACHE[workflowRunId] = parseInt(
+        CONNECTOR_ID_CACHE[workflowRunId] = parseInt(
           described.memo.connectorId,
           10
         );
       }
     }
-    if (!WORKFLOW_ID2CONNECTOR_ID_CACHE[workflowRunId]) {
-      return null;
+  }
+  return CONNECTOR_ID_CACHE[workflowRunId] || null;
+}
+
+export async function getDoNotCancelOnTokenRevoked(
+  workflowRunId: string
+): Promise<boolean> {
+  const cacheHasWorkflowId = Object.prototype.hasOwnProperty.call(
```

Comment:
🤔 

We could probably do `"workflowRunId" in  DO_NOT_CANCEL_ON_TOKEN_REVOKED_CACHE`